integration-tests/tests/otlp-metrics/otlp_metrics_test.go

```diff
@@ -1,23 +1,80 @@
 package main
 
 import (
+	"fmt"
 	"testing"
 
 	"github.com/grafana/agent/integration-tests/common"
 	"github.com/stretchr/testify/assert"
 )
 
-const query = "http://localhost:9009/prometheus/api/v1/query?query=span_metrics_duration_bucket{test_name='otlp_metrics'}"
+const promURL = "http://localhost:9009/prometheus/api/v1/query?query="
 
-func TestOtlpMetrics(t *testing.T) {
+func metricQuery(metricName string) string {
+	return fmt.Sprintf("%s%s{test_name='otlp_metrics'}", promURL, metricName)
+}
+
+func TestOTLPMetrics(t *testing.T) {
+	tests := []struct {
+		query  string
+		metric string
+	}{
+		// TODO: better differentiate these metric types?
+		{metricQuery("example_counter"), "example_counter"},
+		{metricQuery("example_float_counter"), "example_float_counter"},
+		{metricQuery("example_updowncounter"), "example_updowncounter"},
+		{metricQuery("example_float_updowncounter"), "example_float_updowncounter"},
+		{metricQuery("example_histogram_bucket"), "example_histogram_bucket"},
+		{metricQuery("example_float_histogram_bucket"), "example_float_histogram_bucket"},
+	}
+
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.metric, func(t *testing.T) {
+			t.Parallel()
+			assertMetricData(t, tt.query, tt.metric)
+		})
+	}
+	t.Run("example_exponential_histogram", func(t *testing.T) {
+		t.Parallel()
+		assertHistogramData(t, metricQuery("example_exponential_histogram"), "example_exponential_histogram")
+	})
+
+	t.Run("example_exponential_float_histogram", func(t *testing.T) {
+		t.Parallel()
+		assertHistogramData(t, metricQuery("example_exponential_float_histogram"), "example_exponential_float_histogram")
+	})
+}
+
+func assertHistogramData(t *testing.T, query string, expectedMetric string) {
+	var metricResponse common.MetricResponse
+	assert.EventuallyWithT(t, func(c *assert.CollectT) {
+		err := common.FetchDataFromURL(query, &metricResponse)
+		assert.NoError(c, err)
+		if assert.NotEmpty(c, metricResponse.Data.Result) {
+			assert.Equal(c, metricResponse.Data.Result[0].Metric.Name, expectedMetric)
+			assert.Equal(c, metricResponse.Data.Result[0].Metric.TestName, "otlp_metrics")
+			if assert.NotNil(c, metricResponse.Data.Result[0].Histogram) {
+				histogram := metricResponse.Data.Result[0].Histogram
+				assert.NotEmpty(c, histogram.Data.Count)
+				assert.NotEmpty(c, histogram.Data.Sum)
```

Comment:
Is `Count` and `Sum` a number? What does checking that they're empty mean?

Do we want to instead check they're greater than 0, or maybe greater than 10? (they will eventually be greater than 10 from what I gather).

Reply:
count and sum are numbers represented as strings in the answer. Checking that the string is not empty means that there is a number. I can convert it as a number and check that it is > 10 for better precision.

Reply:
ah, I see - yeah, I guess you can can check first if empty, then try convert and pick some number that will be eventually reached :)